api/maven-api-model/src/main/mdo/maven.mdo

```diff
@@ -2646,6 +2646,15 @@
             ]]>
           </description>
         </field>
+        <field>
+          <name>priority</name>
+          <version>4.2.0+</version>
+          <type>int</type>
+          <description>The priority of this execution compared to other executions which are bound to the same phase.
+            Executions derived from the default lifecycle have a negative priority by default so that they are executed
+            before any custom plugin executions.
```

Comment:
nice description
idea: IIRC, priority is set to -1 for lifecycle bindings, where default priority is 0
while introducing this priority, using -1000 instead of -1 would permit users to insert between lifecycle binding and default 0 (and would start to clarify why priority vales is not ideal, but just an easy opportunistic feature to use only when necessary)

note for self: I also need to check that when bindings have been defined with default priority, we can in a child pom just override the priority by just putting execution id

Reply:
The priority for lifecycle plugin executions is negative, but not -1, see https://github.com/apache/maven/blob/master/maven-core/src/main/java/org/apache/maven/lifecycle/internal/DefaultLifecyclePluginAnalyzer.java#L191